test/wpt/status/streams.json

```diff
@@ -0,0 +1,14 @@
+{
+  "piping/pipe-through.any.js": {
+    "fail": "Node does not perform a brand check for AbortSignal.prototype.aborted"

```

Comment:
[This test](https://github.com/web-platform-tests/wpt/blob/887350c2f46def5b01c4dd1f8d2eee35dfb9c5bb/streams/piping/pipe-through.any.js#L176) checks whether passing a "bad" `AbortSignal` (in particular ) causes `pipeThrough()` to throw. [The polyfill implements this](https://github.com/MattiasBuelens/web-streams-polyfill/blob/6915a01aa6c92b4d3cb306fb7a66299e881a7bcc/src/lib/abort-signal.ts#L33-L38) by trying to retrieve `AbortSignal.prototype.aborted` and checking that it does not throw. [WebIDL requires this attribute to perform a brand check](https://heycam.github.io/webidl/#dfn-attribute-getter), so a compliant implementation *should* throw a `TypeError` when this getter is run on such a "bad" `AbortSignal`.

Unfortunately, Node's [AbortSignal implementation](https://github.com/nodejs/node/blob/d15475578abfcc7e0a420719b8d55a6b3ab4f198/lib/internal/abort_controller.js#L46) **does not** throw an error when `AbortSignal.prototype.aborted` is run such a signal. Instead, it will return `false` and the polyfill will consider this to be a valid `AbortSignal`.

I'm not sure what we want to do here. Should we fix Node to perform a brand check in `get aborted` (i.e. throw an error if `this[kAborted]` does not exist)? Or should we change how the polyfill does this check (although I'm not sure how)?

Reply:
If I understand you correctly, I think the best course of action is to make an upstream change to Node.js core AbortSignal so that it aligns more with the spec.

Reply:
On it! 🙂